File: backend/api/internal/domain/content/article.go
package content

import (
	"time"

	"github.com/google/uuid"
)

type ArticleStatus string

const (
	ArticleStatusDraft     ArticleStatus = "draft"
	ArticleStatusPublished ArticleStatus = "published"
	ArticleStatusArchived  ArticleStatus = "archived"
)

// ArticleRevision is an immutable snapshot of an article at a given version.
type ArticleRevision struct {
	ID          uuid.UUID
	ArticleID   uuid.UUID
	Version     int
	EditorID    uuid.UUID
	Title       string
	Body        string
	Slug        string
	Category    string
	Excerpt     string
	ReadingTime int
	Difficulty  string
	Verified    bool
	Status      ArticleStatus
	CreatedAt   time.Time
}

type Article struct {
	ID           uuid.UUID
	Slug         string
	Title        string
	Body         string
	Category     string
	Excerpt      string
	ReadingTime  int
	Difficulty   string
	Verified     bool
	Contributors int
	AuthorID     uuid.UUID
	Status       ArticleStatus
	Version      int
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: apps/admin/src/lib/core/domain/article.ts
export type ArticleStatus = 'draft' | 'published' | 'archived';

export interface Article {
  id: string;
  slug: string;
  title: string;
  body: string;
  category?: string;
  excerpt?: string;
  reading_time?: number;
  difficulty?: string;
  verified?: boolean;
  contributors?: number;
  author_id: string;
  status: ArticleStatus;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface ArticleRevision {
  id: string;
  article_id: string;
  version: number;
  editor_id: string;
  title: string;
  body: string;
  slug: string;
  category: string;
  excerpt?: string;
  reading_time: number;
  difficulty: string;
  verified: boolean;
  status: ArticleStatus;
  created_at: string;
}

export interface ArticleWrite {
  title: string;
  body?: string;
  category?: string;
  excerpt?: string;
  difficulty?: string;
  verified?: boolean;
  status?: ArticleStatus;
}

export interface ArticlePatch {
  status?: ArticleStatus;
  verified?: boolean;
}
